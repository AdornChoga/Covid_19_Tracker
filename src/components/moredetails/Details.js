import React from 'react';
import PropTypes from 'prop-types';
import classes from './MoreDetails.module.css';

const Details = (props) => {
  const { details } = props;
  const {
    todayNewCases, todayRecoveries, todayDeaths, totalCasesRecorded,
    totalRecoveriesRecorded, totalDeathsRecorded,
  } = details;
  return (
    <ul className={classes.statistics}>
      <li>
        Today&apos;s new cases:&nbsp;&nbsp;&nbsp;
        {todayNewCases}
      </li>
      <li>
        Today&apos;s new recoveries:&nbsp;&nbsp;&nbsp;
        {todayRecoveries}
      </li>
      <li>
        Today&apos;s new deaths:&nbsp;&nbsp;&nbsp;
        {todayDeaths}
      </li>
      <li>
        Total cases recorded:&nbsp;&nbsp;&nbsp;
        {totalCasesRecorded}
      </li>
      <li>
        Total recoveries recorded:&nbsp;&nbsp;&nbsp;
        {totalRecoveriesRecorded}
      </li>
      <li>
        Total deaths recorded:&nbsp;&nbsp;&nbsp;
        {totalDeathsRecorded}
      </li>
    </ul>
  );
};

Details.propTypes = {
  details: PropTypes.shape({
    todayNewCases: PropTypes.number,
    todayRecoveries: PropTypes.number,
    todayDeaths: PropTypes.number,
    totalCasesRecorded: PropTypes.number,
    totalRecoveriesRecorded: PropTypes.number,
    totalDeathsRecorded: PropTypes.number,
  }).isRequired,
};

export default Details;
