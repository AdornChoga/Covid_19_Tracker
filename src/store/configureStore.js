import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import statistics from './statistics/statistics';

const reducer = combineReducers({
  statistics,
});

const store = createStore(reducer, applyMiddleware(logger, thunk));

export default store;
