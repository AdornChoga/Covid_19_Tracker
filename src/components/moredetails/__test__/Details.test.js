import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../../store/configureStore';
import Details from '../Details';

it('should create details component snapshot', () => {
  const details = {
    todayNewCases: 1, todayRecoveries: 2, todayDeaths: 3, totalCasesRecorded: 4,
    totalRecoveriesRecorded: 5, totalDeathsRecorded: 6,
  };
  const tree = renderer.create(
  <Provider store={store}>
    <Router>
      <Details details={details} />
    </Router>
  </Provider>);
  expect(tree).toMatchSnapshot();
});