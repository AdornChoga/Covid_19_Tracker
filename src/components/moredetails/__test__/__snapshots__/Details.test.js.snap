// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create details component snapshot 1`] = `
<ul
  className="statistics"
>
  <li>
    Today's new cases:   
    1
  </li>
  <li>
    Today's new recoveries:   
    2
  </li>
  <li>
    Today's new deaths:   
    3
  </li>
  <li>
    Total cases recorded:   
    4
  </li>
  <li>
    Total recoveries recorded:   
    5
  </li>
  <li>
    Total deaths recorded:   
    6
  </li>
</ul>
`;
